### PR TITLE
MSI-2719: Introduce a Bulk version of IsProductSalableForRequestedQtyInterface API.

### PR DESCRIPTION
**Pull request by nmalevanec**
### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2719: Introduce a Bulk version of IsProductSalableForRequestedQtyInterface API

### Manual testing scenarios (*)

1. none.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace Magento\InventorySalesApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Represents result of service Magento\InventorySalesApi\Api\AreProductsSalableForRequestedQtyInterface::execute()
+ *
+ * @api
+ */
+interface AreProductsSalableResultInterface extends ExtensibleDataInterface
```
Comment (sidolov) [inline InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php]: I believe the name `ProductsSalableResultInterface` will be better here.

File: InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+use Magento\Framework\Model\Context;
+use Magento\Framework\Model\ResourceModel\AbstractResource;
+use Magento\Framework\Registry;
+use Magento\InventorySalesApi\Api\Data\AreProductsSalableResultExtensionInterface;
+use Magento\InventorySalesApi\Api\Data\AreProductsSalableResultInterface;
+
+/**
+ * @inheritDoc
+ */
+class AreProductSalableResult extends AbstractExtensibleModel implements AreProductsSalableResultInterface
```
Comment (sidolov) [inline InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php]: Better to use`ProductsSalableResult` name here

File: InventorySales/Model/CheckItemsQuantity.php [hunk 3]
```diff
@@ -7,40 +7,54 @@
...
+ * Verify items are salable for requested quantity.
+ */
 class CheckItemsQuantity
 {
     /**
+     * @deprecated
+     * @see AreProductsSalableForRequestedQty
      * @var IsProductSalableForRequestedQtyInterface
      */
     private $isProductSalableForRequestedQty;
```
Comment (sidolov) [inline InventorySales/Model/CheckItemsQuantity.php]: You may remove private property and constructor assignment if it not used anymore.

File: InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            $resourceCollection,
+            $data
+        );
+        $this->results = $results;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAreSalable(): array
```
Comment (sidolov) [inline InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php]: Please, use `getSalable` method name here

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+{
+    /**
+     * Retrieve is salable results.
+     *
+     * @return \Magento\InventorySalesApi\Api\Data\ProductSalableResultInterface[]
+     */
+    public function getAreSalable(): array;
```
Comment (sidolov) [inline InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php]: Please, use `getSalable` name here

File: InventorySales/etc/di.xml [hunk 6]
```diff
@@ -130,10 +132,15 @@
             </argument>
         </arguments>
     </virtualType>
-    <type name="Magento\InventorySales\Model\CheckItemsQuantity">
+    <virtualType name="AreProductsSalableForRequestedQtyConditionChainOnPlaceOrder" type="Magento\InventorySales\Model\AreProductsSalableForRequestedQty">
```
Comment (sidolov) [inline InventorySales/etc/di.xml]: Is it possible to avoid virtual type here?

**Review by sidolov — CHANGES_REQUESTED** [review]

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php (hunk 1 above)
Comment (nmalevanec) [reply InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php]: @sidolov, renamed

File: InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php (hunk 2 above)
Comment (nmalevanec) [reply InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php]: @sidolov, renamed

File: InventorySales/Model/CheckItemsQuantity.php (hunk 3 above)
Comment (nmalevanec) [reply InventorySales/Model/CheckItemsQuantity.php]: @sidolov, removed

File: InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php (hunk 4 above)
Comment (nmalevanec) [reply InventorySales/Model/AreProductsSalableForRequestedQty/AreProductSalableResult.php]: @sidolov, renamed

File: InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php (hunk 5 above)
Comment (nmalevanec) [reply InventorySalesApi/Api/Data/AreProductsSalableResultInterface.php]: @sidolov, renamed

File: InventorySales/etc/di.xml (hunk 6 above)
Comment (nmalevanec) [reply InventorySales/etc/di.xml]: @sidolov, removed

File: InventorySales/Model/AreProductsSalableForRequestedQty/ProductsSalableResult.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+use Magento\Framework\Model\Context;
+use Magento\Framework\Model\ResourceModel\AbstractResource;
+use Magento\Framework\Registry;
+use Magento\InventorySalesApi\Api\Data\ProductsSalableResultExtensionInterface;
+use Magento\InventorySalesApi\Api\Data\ProductsSalableResultInterface;
+
+/**
+ * @inheritDoc
+ */
+class ProductsSalableResult extends AbstractExtensibleModel implements ProductsSalableResultInterface
```
Comment (sidolov) [inline InventorySales/Model/AreProductsSalableForRequestedQty/ProductsSalableResult.php]: Please, remove the inheritance of `\Magento\Framework\Model\AbstractExtensibleModel` and just implement extension attributes getter and setter. You may find a good example here `\Magento\InventoryInStorePickup\Model\PickupLocation`.
By this change, we will resolve MHI failure.

**Review by sidolov — CHANGES_REQUESTED** [review]

File: InventorySales/Model/AreProductsSalableForRequestedQty/ProductsSalableResult.php (hunk 7 above)
Comment (nmalevanec) [reply InventorySales/Model/AreProductsSalableForRequestedQty/ProductsSalableResult.php]: done.

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento run Functional Tests B2B

**Comment by sidolov** [comment]
@magento run Functional Tests B2B

**Pull request merged**